From 77c8a80e412550f3f47369f6c220de30dfe5664c Mon Sep 17 00:00:00 2001
From: evilwombat <evilwombat@server.fake>
Date: Tue, 6 Aug 2013 21:25:12 -0700
Subject: [PATCH] ARM: Ambarella: Configure GPIOs for SPI0

Add the GPIO configuration entries to put the SPI0 GPIOs
into hardware-control mode, to allow use of the SPI core to
communicate with the front-panel LCD on the Hero2 camera.

Signed-off-by: evilwombat <evilwombat@server.fake>
---
 arch/arm/mach-ambarella/init-generic.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm/mach-ambarella/init-generic.c b/arch/arm/mach-ambarella/init-generic.c
index 5e95730..afc1ce3 100755
--- a/arch/arm/mach-ambarella/init-generic.c
+++ b/arch/arm/mach-ambarella/init-generic.c
@@ -319,6 +319,13 @@ static void __init ambarella_init_generic(void)
 	/* Record button */
 	ambarella_gpio_config(22, GPIO_FUNC_SW_INPUT);
 	ambarella_gpio_config(43, GPIO_FUNC_SW_INPUT);
+
+	/* SPI0, for front panel LCD */
+        ambarella_gpio_config(SSI0_CLK,  GPIO_FUNC_HW);
+        ambarella_gpio_config(SSI0_MOSI, GPIO_FUNC_HW);
+        ambarella_gpio_config(SSI0_MISO, GPIO_FUNC_HW);
+        ambarella_gpio_config(SSI0_EN0,  GPIO_FUNC_HW);
+        ambarella_gpio_config(SSI0_EN1,  GPIO_FUNC_HW);
 }
 
 /* ==========================================================================*/
-- 
1.7.9.5

